File: 9. Week 9 - Maps/exercise/r/visualization.R
# Visualize GeoJSON Maps using ggplot2 and sf

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(sf)) install.packages("sf") # For GeoJSON handling
library(ggplot2)
library(sf)

# We need to make sure the path points to our data file
data_path <- "../../data/runs.geojson"

if (file.exists(data_path)) {
  # Load GeoJSON
  runs_sf <- st_read(data_path, quiet = TRUE)
  
  # Plot
  # TODO: Create the map using ggplot2 and sf
  # Use geom_sf() which handles coordinate projection automatically
  
  # p <- ggplot(runs_sf) +
  #   geom_sf(...)
  
  
  # Uncomment to save
  # ggsave("map.png", plot = p, width = 10, height = 10)
  # print("Map saved to map.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 9. Week 9 - Maps/exercise/python/visualization.py
import json
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/runs.geojson'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load Data
    # with open(data_path, 'r') as f:
    #     data = ...
    
    plt.figure(figsize=(10, 10))
    
    # Iterate through features and plot LineStrings
    # for feature in data['features']:
        # geom = feature['geometry']
        # if geom['type'] == 'LineString':
            # coords = geom['coordinates']
            
            # TODO: Convert GeoJSON [lon, lat] for matplotlib
            # lons = ...
            # lats = ...
            
            # TODO: Plot the path
            # plt.plot(...)
            
    # TODO: Add title and adjust axis
    # plt.title(...)
    # plt.axis('equal') # Standard for maps
    # plt.axis('off')   # Clean look
    
    plt.tight_layout()
    # TODO: Save
    # plt.savefig(...)
    
    print("Map saved to map.png")
    plt.show()

if __name__ == "__main__":
    visualize()
